### PR TITLE
Assigning index task with fair queue

### PR DESCRIPTION
**Pull request by xiaocai2333**
/kind improvement

### CODE REVIEW

File: internal/datacoord/queue.go [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+	}
+	oldState := task.state
+	rtq.tasks[collectionID][buildID].state = state
+	log.Info("update task state success", zap.Int64("instanceID", collectionID),
+		zap.Int64("buildID", buildID), zap.String("old state", oldState.String()),
+		zap.String("new state", state.String()))
+}
+
+func (rtq *runningTasksQueue) getTaskState(collectionID, buildID UniqueID) indexTaskState {
+	rtq.lock.Lock()
```
Comment (czs007) [inline internal/datacoord/queue.go:291]: RLock?

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request closed without merging**